provisioning/miniprov/hare_mp/main.py

```diff
@@ -243,11 +251,48 @@ def is_cluster_running() -> bool:
     return os.system('hctl status >/dev/null') == 0
 
 
+def nr_services() -> int:
+    cmd = ['hctl', 'status', '--json']
+    cluster_info = json.loads(execute(cmd))
+    # Don't include hax, count it just once later.
+    services = {'confd', 'ioservice', 's3server'}
+    svcs_nr = 0
+    for node in cluster_info['nodes']:
+        for svc in node['svcs']:
+            if svc['name'] in services:
+                svcs_nr += 1
+    return svcs_nr + 1
+
+
+def all_services_started(url: str, nr_svcs: int) -> bool:
```

Comment:
`@repeat_if_fails()` is probably missed. In the current implementation, HAConsistencyException will cause the script to exit early.

Reply:
Good point!